SPECS-SIGNED/kernel-mos-signed/kernel-mos-signed.spec

```diff
@@ -0,0 +1,153 @@
+# spec or the normal kernel-mos spec, the other spec's version version/release must
+# be increased to keep the two versions consistent.
+#
+# - Make sure the kernel-mos subpackage's Requires, Provides, triggers, post/postun
+# scriptlets, and files match the normal kernel-mos spec's. The kernel subpackage
+# should contain the same content as the input kernel-mos package but replace the
+# kernel binary with our signed kernel binary. Since all the requires, provides,
+# etc are the same, this new kernel-mos package can be a direct replacement for the
+# normal kernel-mos package and RPM will resolve packages with kernel-mos dependencies
+# correctly.
+#
+# To populate the input sources:
+#   1. Build the unsigned packages as normal
+#   2. Sign the desired binary
+#   3. Place the unsigned package and signed binary in this spec's folder
+#   4. Build this spec
+Source0:        kernel-mos-%{version}-%{release}.%{buildarch}.rpm
+Source1:        vmlinuz-%{uname_r}
+Source2:        sha512hmac-openssl.sh
+BuildRequires:  cpio
+BuildRequires:  openssl
+BuildRequires:  sed
+
+%description
+This package contains the Linux kernel package with kernel signed with the production key
+
+%package -n     kernel-mos
+Summary:        Linux Kernel for MOS
+Group:          System Environment/Kernel
+Requires:       filesystem
+Requires:       kmod
+Requires(post): coreutils
+Requires(postun): coreutils
+
+%description -n kernel-mos
+The kernel-mos package contains the signed Linux kernel for MOS.
+
+%prep
+
+%build
+mkdir rpm_contents
+pushd rpm_contents
+
+# This spec's whole purpose is to inject the signed kernel binary
+rpm2cpio %{SOURCE0} | cpio -idmv
+cp %{SOURCE1} ./boot/vmlinuz-%{uname_r}
+
+popd
+
+%install
+pushd rpm_contents
+
+# Don't use * wildcard. It does not copy over hidden files in the root folder...
+cp -rp ./. %{buildroot}/
+
+popd
+
+# Recalculate sha512hmac for FIPS
+%{sha512hmac} %{buildroot}/boot/vmlinuz-%{uname_r} | sed -e "s,$RPM_BUILD_ROOT,," > %{buildroot}/boot/.vmlinuz-%{uname_r}.hmac
+cp %{buildroot}/boot/.vmlinuz-%{uname_r}.hmac %{buildroot}/lib/modules/%{uname_r}/.vmlinuz.hmac
+
+%triggerin -n kernel-mos -- initramfs
+mkdir -p %{_localstatedir}/lib/rpm-state/initramfs/pending
+touch %{_localstatedir}/lib/rpm-state/initramfs/pending/%{uname_r}
+echo "initrd generation of kernel %{uname_r} will be triggered later" >&2
+
+%triggerun -n kernel-mos -- initramfs
+rm -rf %{_localstatedir}/lib/rpm-state/initramfs/pending/%{uname_r}
+rm -rf /boot/initrd.img-%{uname_r}
+echo "initrd of kernel %{uname_r} removed" >&2
+
+%postun -n kernel-mos
+if [ ! -e /boot/mariner.cfg ]
+then
+     ls /boot/linux-*.cfg 1> /dev/null 2>&1
+     if [ $? -eq 0 ]
+     then
+          list=`ls -tu /boot/linux-*.cfg | head -n1`
+          test -n "$list" && ln -sf "$list" /boot/mariner.cfg
+     fi
+fi
+
+%post -n kernel-mos
+/sbin/depmod -a %{uname_r}
+ln -sf linux-%{uname_r}.cfg /boot/mariner.cfg
+
+%files -n kernel-mos
```

Comment:
nit - this `kernel-mos` `%files` should ideally match the `kernel-mos.spec`'s main `kernel-mos` package's `%files`

Given the diff is only additional `%exclude` entries in the `kernel-mos.spec`'s `%files`, there isn't an end difference. So this is mostly a nit to keep them in sync for spec tidiness reasons. And to make sure our tests in our official build are active to catch for actual regressions.

Reply:
Saw the update - just did a diff of the %files between `kernel-mos` and these %files, and there are still differences

Reply:
I updated files to use a macro to match the kernel-mos spec. However, I kept the `exclude module_info.ld ` in only kernel-mos-signed which is intentionally excluded in "kernel-signed" as it was causing build breaks https://github.com/microsoft/CBL-Mariner/pull/1472 .